### ISSUE TITLE
Discrepancy in line wrapping when rendering chat preview

### ISSUE THREAD

**Issue by Robyt3** [ddnet/ddnet#8179]
Set the chat font size to 100 and the chat width to 262, 263 and 264 respectively:

- ![screenshot_2024-04-02_21-35-29](https://github.com/ddnet/ddnet/assets/23437060/aea56055-2253-4689-a247-cd703b5969aa)
- ![screenshot_2024-04-02_21-35-26](https://github.com/ddnet/ddnet/assets/23437060/4bba65a3-4a2b-4f65-a928-cc515f673264)
- ![screenshot_2024-04-02_21-35-24](https://github.com/ddnet/ddnet/assets/23437060/153beaa4-a1ba-4515-816e-2759ad737242)

With chat width 263 one of the background rectangles has an incorrect size which assumes the text wraps to the next line.

Haven't tested if this also extends to regular chat rendering.

**Comment by archimede67** [comment]
I don't think this extends to regular chat rendering. I have encountered this bug when I made that feature, I thought I had fixed it but it seems like there are some edge cases. The rendering code in the menu is not the same as the one used ingame, though IIRC I took inspiration from it.

**Comment by gerdoe-jr** [comment]
Can't reproduce with all possible values

**Comment by Robyt3** [comment]
It likely depends on your player name and maybe also the screen size. It definitely happens with `nameless tee` and 1920x1080.

**Comment by Robyt3** [comment]
Might be the result of differences in floating point error due to the `TextWidth` function "rendering" the text internally at position `(0.0f, 0.0f)` whereas the real text rendering will start at a difference position, which results in slightly different glyph positions.

**Comment by Robyt3** [comment]
Also reported to be an issue ingame by cyberFighter on Discord:

![image](https://github.com/user-attachments/assets/f8c27006-74ce-4b68-b5e0-081a469190eb)
